doc/source/development/dispatching.rst

```diff
@@ -0,0 +1,98 @@
+Dispatching
+===========
+
+The dispatching mechanism allows users to use alternative implementations of the algorithms
+available in scikit-image. The alternative implementations are provided by separate
+Python packages maintained outside of scikit-image. Example use-cases are support for array
+libraries other than Numpy or optimised implementations for particular hardware.
+
+This section of the documentation describes how to create an alternative implementation (a "backend")
+and how the dispatching mechanism in scikit-image works.
+
+.. important::
+    The dispatching API is experimental and is not ready for production.
+    It is made available as an early prototype so that developers can gain experience
+    with the system.
+
+    Consider the dispatching API to be very unstable; we might change behavior
+    suddenly between releases without a deprecation period.
+
+
+Creating a scikit-image backend
+-------------------------------
+
+An alternative implementation ("backend") is a good place to provide optimized implementations
+for particular hardware, support array libraries other than Numpy or explore novel ideas that
+are not (yet) a good fit for the core scikit-image library.
+
+To create a backend you have to create a new Python package that registers two particular
+`entry points <https://packaging.python.org/en/latest/specifications/pyproject-toml/#entry-points>`_.
+:ref:`backend-infos-entry-point` should resolve to a function with no
+arguments that returns an instance of the :py:class:`skimage.util._backends.BackendInformation` class
+or something that behaves like it.
+:ref:`backend-entry-point` should resolve to a namespace
+that contains two functions: ``can_has(name, *args, **kwargs)`` and ``get_implementation(name)``.
+
+.. _backend-infos-entry-point:
+
+The `skimage_backend_infos` entry point
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The information your backend provides via this entry point is used to get things like its
+name, homepage and which functions it implements. The main requirement for this entry point
+is that it is fast to import and has no dependencies other than scikit-image.
+
+The return value of the function should be an instance of the
+``skimage.util._backends.BackendInformation`` class or something that behaves like it.
+
+The reason this entry point has to be fast is that the list of implemented functions
+it provides is used to make a decision on which backends to try for a particular scikit-image
```

Comment:
```suggestion
is used to make a decision on which backends to try for a particular scikit-image
```
Not a big deal, I also thought about suggesting "... the list of functions it implements" because that's the wording everywhere else, but basically it's clear enough with "... the list of implemented functions."